File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function isPrime(n) {
    if (n <= 1) return false;
    if (n === 2) return true;
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) return false;
    }
    return true;
}

function solution(n, k) {
    const arr = n.toString(k).split('0');
    
    return arr.filter(v => v !== "" && isPrime(Number(v))).length;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def convert(n, k):
    s = ''
    while n:
        s += str(n % k)
        n //= k
    return s[::-1]

def isprime(n):
    if n <= 1: return False
    i = 2
    while i*i <= n:
        if n % i == 0: return False
        i += 1
    return True

def solution(n, k):
    s = convert(n, k)
    answer = 0
    for num in s.split('0'):
        if not num: continue
        if isprime(int(num)): answer += 1

    return answer